hazelcast-wm/src/test/java/com/hazelcast/wm/test/WebFilterClientFailOverTests.java

```diff
@@ -0,0 +1,97 @@
+        CookieStore cookieStore = new BasicCookieStore();
+        assertEquals("true", executeRequest("write", serverPort1, cookieStore));
+        assertEquals("value", executeRequest("read", serverPort1, cookieStore));
+        hz.shutdown();
+        assertEquals("value", executeRequest("read", serverPort1, cookieStore));
+        assertEquals("true", executeRequest("remove", serverPort1, cookieStore));
+        assertEquals("null", executeRequest("read", serverPort1, cookieStore));
+        hz = Hazelcast.newHazelcastInstance(
+                new FileSystemXmlConfig(new File(sourceDir + "/WEB-INF/", "hazelcast.xml")));
+        Thread.sleep(8000);
+        assertEquals("true", executeRequest("write", serverPort1, cookieStore));
+        assertEquals("value", executeRequest("read", serverPort1, cookieStore));
+    }
+
+    @Test
+    public void whenClusterIsDownAtBeginning_enabledDeferredWrite() throws Exception {
+        hz.shutdown();
+        CookieStore cookieStore = new BasicCookieStore();
+        assertEquals("true", executeRequest("write", serverPort1, cookieStore));
+        assertEquals("value", executeRequest("read", serverPort1, cookieStore));
+
+        assertEquals("value", executeRequest("read", serverPort1, cookieStore));
+        assertEquals("true", executeRequest("remove", serverPort1, cookieStore));
+        assertEquals("null", executeRequest("read", serverPort1, cookieStore));
+
+        assertEquals("true", executeRequest("write", serverPort1, cookieStore));
+        assertEquals("value", executeRequest("read", serverPort1, cookieStore));
+        hz = Hazelcast.newHazelcastInstance(
+                new FileSystemXmlConfig(new File(sourceDir + "/WEB-INF/", "hazelcast.xml")));
+        Thread.sleep(8000);
+        assertEquals("value", executeRequest("read", serverPort1, cookieStore));
+        assertEquals("true", executeRequest("remove", serverPort1, cookieStore));
+        assertEquals("null", executeRequest("read", serverPort1, cookieStore));
+    }
+
+    @Test
+    public void whenClusterIsDownAtBeginning_MapSizeAfterClusterIsUp_deferredWrite() throws Exception {
+        if (!testName.equals("client - not deferred")) return;
+        hz.shutdown();
+        CookieStore cookieStore = new BasicCookieStore();
+        assertEquals("true", executeRequest("write", serverPort1, cookieStore));
+        assertEquals("value", executeRequest("read", serverPort1, cookieStore));
+
+        hz = Hazelcast.newHazelcastInstance(
+                new FileSystemXmlConfig(new File(sourceDir + "/WEB-INF/", "hazelcast.xml")));
+        Thread.sleep(8000);
+        assertEquals("value", executeRequest("read", serverPort1, cookieStore));
+        IMap<String, Object> map = hz.getMap(DEFAULT_MAP_NAME);
+        assertEquals(1, map.size());
+    }
+
+    @Test
+    public void whenClusterIsDownAtBeginning_MapSizeAfterClusterIsUp_NonDeferredWrite() throws Exception {
```

Comment:
The name of the test doesn't explain what is happening. So it isn't providing a lot of value.